digi/xbee/devices.py

```diff
@@ -3148,6 +3154,31 @@ def __build_expldata_packet(self, remote_xbee_device, data, src_endpoint, dest_e
                                         x16addr, src_endpoint, dest_endpoint,
                                         cluster_id, profile_id, 0, transmit_options, rf_data=data)
 
+    def __get_actual_mode(self):
+        """
+        Gets and returns the actual operating mode of the XBee device reading the ``AP`` parameter in AT command mode.
+
+        Returns:
+             :class:`.OperatingMode`. The actual operating mode of the XBee device or ``OperatingMode.UNKNOWN`` if the
+                mode could not be read.
+        """
+        try:
+            # Clear the serial input stream.
+            self._serial_port.flushInput()
+            # Send the 'AP' command.
+            self._serial_port.write("ATAP\r".encode("utf-8"))
+            time.sleep(0.1)
+            # Read the 'AP' answer.
+            ap_answer = self._serial_port.read_existing().decode("utf-8").rstrip()
+            if len(ap_answer) == 0:
+                return OperatingMode.UNKNOWN
+            # Return the corresponding operating mode for the AP answer.
+            return OperatingMode.get(int(ap_answer, 16))
+        finally:
+            # Exit AT command mode.
+            self._serial_port.write("ATCN\r".encode("utf-8"))
+            time.sleep(self.__TIMEOUT_ENTER_COMMAND_MODE)
```

Comment:
Wouldn't it be better to create a specific method to exit command mode?

Reply:
I'll move it to a different method.